app/models/order.rb

```diff
@@ -18,11 +18,16 @@ def set_defaults
     self.state = "Created"
   end
 
-  def finalize_order
+  def transition_state
     item_states = order_items.collect(&:state)
+
     if item_states.include?('Failed')
```

Comment:
Does it maybe make sense to convert this `if` block to a private method? Then we would simply do:
`self.state = my_private_method`

Reply:
Yeah I like that idea, updated.